app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,34 @@
+class CommentsController < ApplicationController
+
+  def create
+    case params[:comment_type]
+    when "Question"
+      comment_target = Question.find(params[:question_id])
+    when "Answer"
+      comment_target = Answer.find(params[:answer_id])
+    end
+    @comment = comment_target.comments.build(comment_params)
+    @comment.user_id = current_user.id
+    if @comment.save
+      redirect_to :back
+    else
+      redirect_to :back
+    end
+
+  end
+
+  def destroy
+    comment = Comment.find(params[:id])
+    if comment.destroy
+      redirect_to :back
+    else
+      "???"
```

Comment:
What indeed? This is unlikely to fail - you would have got ActiveRecordNotFound (a 404 in production mode) if the comment didn't exist, and there are no referential integrity constraints. You could just show an error here in the rare case it might fail.